Libraries/xcdriver/Sources/VersionAction.cpp

```diff
@@ -60,36 +62,56 @@ Run(process::Context const *processContext, Filesystem const *filesystem, Option
+        auto values = plist::Dictionary::New();
+
         if (target->version()) {
-            printf("SDKVersion: %s\n", target->version()->c_str());
+            values->set("SDKVersion", plist::String::New(*target->version()));
         }
-        printf("Path: %s\n", target->path().c_str());
+        values->set("Path", plist::String::New(target->path()));
         if (target->platform()->version()) {
-            printf("PlatformVersion: %s\n", target->platform()->version()->c_str());
+            values->set("PlatformVersion", plist::String::New(*target->platform()->version()));
         }
-        printf("PlatformPath: %s\n", target->platform()->path().c_str());
+        values->set("PlatformPath", plist::String::New(target->platform()->path()));
         if (auto product = target->product()) {
             if (product->buildVersion()) {
-                printf("ProductBuildVersion: %s\n", product->buildVersion()->c_str());
+                values->set("ProductBuildVersion", plist::String::New(*product->buildVersion()));
             }
             if (product->copyright()) {
-                printf("ProductCopyright: %s\n", product->copyright()->c_str());
+                values->set("ProductCopyright", plist::String::New(*product->copyright()));
             }
             if (product->name()) {
-                printf("ProductName: %s\n", product->name()->c_str());
+                values->set("ProductName", plist::String::New(*product->name()));
             }
             if (product->userVisibleVersion()) {
-                printf("ProductUserVisibleVersion: %s\n", product->userVisibleVersion()->c_str());
+                values->set("ProductUserVisibleVersion", plist::String::New(*product->userVisibleVersion()));
             }
             if (product->version()) {
-                printf("ProductVersion: %s\n", product->version()->c_str());
+                values->set("ProductVersion", plist::String::New(*product->version()));
+            }
+        }
+
+        if (options.actions().empty()) {
+            fprintf(stdout,
+                "%s - %s (%s)\n",
+                target->bundleName().c_str(),
+                target->displayName().value_or(target->bundleName()).c_str(),
+                target->canonicalName().value_or(target->bundleName()).c_str());
+
+            for (size_t n = 0; n < values->count(); n++) {
+                if (plist::String const *value = values->value<plist::String>(n)) {
+                    fprintf(stdout, "%s: %s\n", values->key(n).c_str(), value->value().c_str());
+                }
+            }
+
+            fprintf(stdout, "\n");
+        } else {
+            for (std::string const &action : options.actions()) {
+                if (plist::String const *value = values->value<plist::String>(action)) {
+                    fprintf(stdout, "%s\n", value->value().c_str());
+                }
             }
         }
-        printf("\n");
```

Comment:
By default it prints out the SDK then a blank line; with a specific item it doesn't print the blank line. I think that's to support piping the specific item output.